sqlglot/dialects/presto.py

```diff
@@ -626,3 +626,27 @@ def create_sql(self, expression: exp.Create) -> str:
             if kind == "VIEW" and schema.expressions:
                 expression.this.set("expressions", None)
             return super().create_sql(expression)
+
+        def delete_sql(self, expression: exp.Delete) -> str:
+            """
+            Presto only support DELETE FROM a table, without alias, so we need to remove the unnecessary parts.
+            If the original DELETE statement contains more than one table to be deleted, just pick the first one.
+            """
+
+            tables = expression.args.get("tables")
+            if tables:
+                target_table = tables[0]
+                expression = expression.copy()
+                del expression.args["tables"]
+                expression.set("this", target_table)
+
+            table = expression.args.get("this")
+            if isinstance(table, exp.Table):
+                table_alias = table.alias
+                if table_alias:
+                    del table.args["alias"]
+                    for column in expression.find_all(exp.Column):
+                        if column.table == table_alias:
+                            del column.args["table"]
```

Comment:
I'm not sure if getting rid of the `table` like this is safe / correct. Eg. what happens if there's a db or catalog? We would probably need to do `expression.transform(transforms.unqualify_columns)` on the filter child.

Reply:
I dont know what happens if there's a db or catalog. Can you explain it, please?

I think each column belongs to a table, so a column only have a table_alias (or table name) stored into column.table property. I will correct the table name case.

 I am understand the `transforms.unqualify_columns` will assign table alias into the column, but if the transformation work like this, it does not help anything in this case.

Reply:
> I dont know what happens if there's a db or catalog. Can you explain it, please?

Previously we were only popping off the column's `table`, and I was wondering if this could lead to an inconsistent state where a column that original had a fully-qualified name (`catalog`, `db`, `table`, `this`) would only get its `table` removed.

This probably can't happen for the columns of the table we're deleting from, since you'd qualify the columns of that table using only the alias, so there's no catalog/db. But it could happen for columns of other tables, eg. referenced in a subquery inside of the `WHERE` clause:

```sql
DELETE FROM db.t1 AS t1 WHERE NOT t1.c IN (SELECT db.t2.c FROM db.t2 LEFT JOIN db.t3 USING (id));
```

If you also unqualify `db.t2.c` here, that could make the subquery ambiguous, e.g. if `c` existed in both `t2` and `t3`. The current approach with `unqualify_columns` has this issue.

Reply:
Thanks, I got it. 

> Btw, the current approach will laso unqualify db.t2.c here, which could make the subquery ambiguous, e.g. if c existed in both t2 and t3.

Is there any function to fully qualify the catalog/db/table to the column? After that we can just remove the alias and thing should work.

Reply:
Since this is only used to aid the transpilation, which is best-effort, I think I'd just keep what you have but move the `unqualify_columns` transformation inside the branch that checks if there is an alias. We don't want to always unqualify columns, right? We only need it if the target table is aliased.

Reply:
cc @tobymao can you also take a look?

Reply:
@georgesittas Should we always remove table alias, and always keep the fully qualified column?

Prototype code seems to my first version:
```python
table_alias = table.alias
if table_alias:
    del table.args["alias"]

    for column in expression.find_all(exp.Column):
        if column.table == table_alias and not column.catalog and not column.db:
            del column.args["table"]
```

Reply:
https://github.com/tobymao/sqlglot/pull/3487